examples/oft-solana/tasks/solana/setAuthority.ts

```diff
@@ -0,0 +1,169 @@
+ */
+const getOftStore = (programId: string, escrow: string) => {
+    const oftDeriver = new OftPDA(publicKey(programId))
+    const escrowPK = publicKey(escrow)
+    const [oftStorePda] = oftDeriver.oftStore(escrowPK)
+    return oftStorePda
+}
+
+/**
+ * Get the string representation of the authority type.
+ * @param authorityType {AuthorityType}
+ */
+const getAuthorityTypeString = (authorityType: AuthorityType) => {
+    switch (authorityType) {
+        case AuthorityType.MintTokens:
+            return 'MintTokens'
+        case AuthorityType.FreezeAccount:
+            return 'FreezeAccount'
+        default:
+            throw Error(`Unknown authority type: ${authorityType}`)
+    }
+}
+
+// Define a Hardhat task for creating and setting a new Mint/Freeze Authority
+// for OFT on Solana
+// * Create SPL Multisig account for mint authority
+// * Sanity check the new Multisig account
+// * Set Mint Authority
+// * Set Freeze Authority
+// Note:  Only supports SPL Token Standard.
+task('lz:oft:solana:setauthority', 'Mints new SPL Token and creates new OFT Store account')
+    .addParam('eid', 'Solana mainnet or testnet', undefined, devtoolsTypes.eid)
+    .addParam('mint', 'The Token mint public key (used for MABA only)')
+    .addParam('programId', 'The OFT Program id')
+    .addParam('escrow', 'The OFT escrow public key')
+    .addParam('additionalMinters', 'Comma-separated list of additional minters')
+    .addParam(
+        'tokenProgram',
+        'The Token Program public key (used for MABA only)',
+        TOKEN_PROGRAM_ID.toBase58(),
+        devtoolsTypes.string
+    )
+    .setAction(
+        async ({
+            eid,
+            escrow: escrowStr,
+            mint: mintStr,
+            programId: programIdStr,
+            tokenProgram: tokenProgramStr,
+            additionalMinters: additionalMintersStr,
+        }: CreateOFTTaskArgs) => {
+            const { connection, umi, umiWalletKeyPair, umiWalletSigner } = await deriveConnection(eid)
+            const oftStorePda = getOftStore(programIdStr, escrowStr)
+            const tokenProgram = publicKey(tokenProgramStr)
+            const additionalMinters = additionalMintersStr.split(',').map((minter) => new PublicKey(minter))
+            const mint = new PublicKey(mintStr)
+            const newMintAuthority = await createMintAuthorityMultisig(
+                connection,
+                toWeb3JsKeypair(umiWalletSigner),
+                new PublicKey(oftStorePda.toString()),
+                new PublicKey(tokenProgram.toString()),
+                additionalMinters
+            )
+            console.log(`New Mint Authority: ${newMintAuthority.toBase58()}`)
+            const signers = await checkMultisigSigners(connection, newMintAuthority, [
+                toWeb3JsPublicKey(oftStorePda),
+                ...additionalMinters,
+            ])
+            console.log(`New Mint Authority Signers: ${signers.map((s) => s.toBase58()).join(', ')}`)
```

Comment:
this seems to give wrong output despite multisig being properly created on mainnet

Reply:
Fixed.